source/jormungandr/jormungandr/street_network/parking/augeas.py

```diff
@@ -0,0 +1,85 @@
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+import requests, pybreaker, logging
+from jormungandr.street_network.parking.abstract_parking_module import AbstractParkingModule
+
+
+class Augeas(AbstractParkingModule):
+    def __init__(self, service_url, max_park_duration=1200):
+        self.service_url = service_url
+        self.max_park_duration = max_park_duration
+        # TODO: put into a config when the POC is validated
+        self.breaker = pybreaker.CircuitBreaker(fail_max=500, reset_timeout=60)
+        self.logger = logging.getLogger(__name__)
+
+    def _breaker_wrapper(self, fun, *args, **kwargs):
+        try:
+            return self.breaker.call(fun, *args, **kwargs)
+        except Exception as e:
+            self.logger.exception('error occurred when requesting Augeas: {}'.format(str(e)))
+            raise
+
+    def _request_in_batch(self, coords):
+        data = {
+            "n": 1,
+            "walking_speed": 1.11,
+            "max_park_duration": self.max_park_duration,
+            "coords": [[c.lon, c.lat] for c in coords],
+        }
+        url = requests.compat.urljoin(self.service_url, '/v0/park_duration')
+        r = requests.post(url=url, json=data)
+        return r.json().get('durations')
+
+    def _request(self, coord):
+        params = {
+            'lon': coord.lon,
+            'lat': coord.lat,
+            'n': 1,
+            "walking_speed": 1.11,
+            'max_park_duration': self.max_park_duration,
+        }
+        url = requests.compat.urljoin(self.service_url, '/v0/park_duration')
+        r = requests.get(url=url, params=params)
+        durations = r.json().get('durations')
+        if not durations:
+            return self.max_park_duration
+        return durations[0].get('duration', self.max_park_duration)
+
+    def get_parking_duration(self, coord):
+        return self._breaker_wrapper(self._request, coord=coord)
+
+    def get_leave_parking_duration(self, coord):
+        return self._breaker_wrapper(self._request, coord=coord)
```

Comment:
what is the difference with `get_parking_duration` ? they seem pretty similar. (same for the 2 functions above)

Reply:
you're right, they're the same in the implementation, even though they're different in real life, it should be handled differently later